tests/rptest/tests/datalake/transactions_test.py

```diff
@@ -0,0 +1,139 @@
+
+    def append_last_committed_transaction(self):
+        produced_offset = -1
+
+        def delivery_callback(err, msg):
+            if err:
+                assert False, "failed to deliver message: %s" % err
+            nonlocal produced_offset
+            produced_offset = msg.offset()
+
+        producer = ck.Producer({
+            'bootstrap.servers': self.redpanda.brokers(),
+            'transactional.id': 'last_committed_tx',
+        })
+        producer.init_transactions()
+        producer.begin_transaction()
+        producer.produce(self.topic_name,
+                         "test",
+                         "test",
+                         callback=delivery_callback)
+        producer.flush()
+        producer.commit_transaction()
+        assert produced_offset > 0, f"producer not flushed, offset: {produced_offset}"
+        return produced_offset
+
+    @cluster(num_nodes=4)
+    @matrix(cloud_storage_type=supported_storage_types(),
+            compaction=[True, False])
+    def test_with_transactions(self, cloud_storage_type, compaction):
+        """Test ensures correctness of translation when running with transactions.
+        Ensures no aborted transactions make it to the iceberg table"""
+
+        tx_verifier = CompactedVerifier(self.test_context, self.redpanda,
+                                        Workload.TX)
+        min_num_records = 1000
+        with DatalakeServices(self.test_context,
+                              redpanda=self.redpanda,
+                              filesystem_catalog_mode=False,
+                              include_query_engines=[QueryEngineType.TRINO
+                                                     ]) as dl:
+            topic_config = dict()
+            topic_config[TopicSpec.PROPERTY_SEGMENT_SIZE] = 1 * 1024 * 1024
+            if compaction:
+                topic_config[
+                    TopicSpec.
+                    PROPERTY_CLEANUP_POLICY] = TopicSpec.CLEANUP_COMPACT
+            dl.create_iceberg_enabled_topic(self.topic_name,
+                                            config=topic_config)
+
+            tx_verifier.start()
+            # Produces transactional data into the topic with 30% aborted batches
+            tx_verifier.remote_start_producer(self.redpanda.brokers(),
+                                              self.topic_name,
+                                              partitions=1)
+            tx_verifier.ensure_progress(delta=min_num_records, timeout_sec=180)
+            tx_verifier.remote_stop_producer()
+            tx_verifier.remote_wait_producer()
+            # Issue a last committed transaction so we do not end with an aborted
+            # transaction which makes the verification difficult (as we need to keep
+            # track of last committed transaction offset)
+            offset = self.append_last_committed_transaction()
+            tx_verifier.stop()
```

Comment:
Just curious, does it matter the order of stopping and appending the last committed transaction?